### PR TITLE
Boarding Event Update

### PR DESCRIPTION
**Pull request by IlyaIU**
Adds Basic stuff for boarding event.
For the testing time, players can't trigger boarding event ingame. It can be triggered by admins in two ways:

- Pressing the Test Boarding button in the FTL tab
- Spawning Test Ship with a boarding capability

Im handling tests by myself, i'll say when it's ready.

TODO list:
- [x] Wrecks
- [x] New ghost role
- [x] Starship map for boarding
- [x] Point capture mode
- [x] Debug procs
- [ ] Playtests

:cl: injazz
add: adds the button for admin to call boarding event
wip: boarding event - capture the enemy ship's vault and have nice things
wip: ghost role - ship's defender - don't let those nanotrasens have nice things
tweak: now mapping datum can make additional zlevels on top of the existing ones
/:cl:

### CODE REVIEW

**Comment by IlyaIU** [comment]
it's so heavy wip i didn't even compiled it
i want to sleep

File: code/game/gamemodes/miniantags/ftl_boarding/_init.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+They got a spare control program hidden in the remaining spare terminal(aka Point). Also they got some TC installed in there. In order for termial to
+work it should be anchored somewhere, so you would manage TC to get some shit to defend themselves.
+Player ship crew (aka Atk) should rush into the wreckage and capture the Point or just murderbone every enemy crewmen they will encounter.
+Atk wins = Atk got access to Vault and can have some nice things from it, remaining Def explodes violently.
+Def wins = ship explodes into the pieces, everyone involved dies. VIOLENTLY.
+*/
+
+//Loading boarding map
+/datum/subsystem/starmap/proc/init_boarding(var/datum/starship/S)
+  var/datum/round_event/ghost_role/boarding/mode = new /datum/ftl_boarding
```
Comment (IlyaIU) [inline code/game/gamemodes/miniantags/ftl_boarding/_init.dm]: oh fuck

**Comment by IlyaIU** [comment]
Current status:
- [ ] No ship maps, using placeholder for now (im not a mapper)
- [ ] No defence gear
- [ ] No point capture restrictions (gibbing defenders who left the ship for example)

**Comment by jakmak6** [comment]
I would reccomend looking at porting /vg/ holomaps for the boarding parties since that kind of utility would make boarding less of a disorganized mess

**Comment by IlyaIU** [comment]
@monster860  merge and use build_assets.bat for tgui

**Comment by ike709** [comment]
You say merge but travis says don't merge.

**Pull request merged**